ibc/packet.go

```diff
@@ -0,0 +1,82 @@
+package ibc
+
+import (
+	"errors"
+	"fmt"
+	"time"
+
+	host "github.com/cosmos/ibc-go/v3/modules/core/24-host"
+)
+
+// Packet is a packet sent over an IBC channel as defined in ICS-4.
+// See: https://github.com/cosmos/ibc/blob/master/spec/core/ics-004-channel-and-packet-semantics/README.md
+// Proto defined at: github.com/cosmos/ibc-go/v3@v3.0.0/proto/ibc/core/channel/v1/tx.proto
+type Packet struct {
+	Sequence         uint64        // the order of sends and receives, where a packet with an earlier sequence number must be sent and received before a packet with a later sequence number
+	SourcePort       string        // the port on the sending chain
+	SourceChannel    string        // the channel end on the sending chain
+	DestPort         string        // the port on the receiving chain
+	DestChannel      string        // the channel end on the receiving chain
+	Data             []byte        // an opaque value which can be defined by the application logic of the associated modules
+	TimeoutHeight    string        // a consensus height on the destination chain after which the packet will no longer be processed, and will instead count as having timed-out
+	TimeoutTimestamp time.Duration // indicates a timestamp on the destination chain after which the packet will no longer be processed, and will instead count as having timed-out
```

Comment:
Good comments on these struct fields.

Is there a more specific definition of TimeoutTimestamp? Since it is a time.Duration, is the actual timeout calculated from when the packet is Received by the Chain?

Reply:
Ha, I plagurized the IBC spec documents for these comments. I have a future vision where `ibctest` can become a great learning source for IBC itself.

I used `time.Duration` because the timeout is nanoseconds which is also `time.Duration`'s unit. I'll update the comment.

In Go, I typically don't like using anything other than `time.Duration` to represent durations. All too often you see `int` or similar scalars which is error prone.

Reply:
Oh wait, I need more coffee apparently. This is a *timestamp* not a duration. Yes, I'll re-eval this field. It should be nanoseconds (presumably since epoch). I'll dig more into it.